### ISSUE TITLE
Error filtering on documents with values ending with a single backslash

### ISSUE THREAD

**Issue by benjamin-cizej** [meilisearch/meilisearch#4034]
**Describe the bug**
Meilisearch returns an error when trying to use a value such as `foo\` in filter parameter.

**To Reproduce**

1. Insert documents - `test_1` document with `foo\` title, `test_2` document with `foo\\` title
POST `/indexes/test/documents`
```
[
    {
        "id": "test_1",
        "title": "foo\\"
    },
    {
        "id": "test_2",
        "title": "foo\\\\"
    }
]
``` 
2. Set filterable `title` attribute
POST `/indexes/test/settings`
```
{
    "filterableAttributes": [
        "title"
    ]
}
``` 
3. Execute search with filter `foo\` to get `test_1` document
POST `/indexes/test/search`
```
{
    "filter": [
        "title = \"foo\\\""
    ]
}
``` 

4. An error is returned
```
{
    "message": "Expression `\\\"foo\\\\\\\"` is missing the following closing delimiter: `\"`.\n9:15 title = \"foo\\\"",
    "code": "invalid_search_filter",
    "type": "invalid_request",
    "link": "https://docs.meilisearch.com/errors#invalid_search_filter"
}
``` 

Same error is returned if I use single quotes in expression.

POST `/indexes/test/search`
```
{
    "filter": [
        "title = 'foo\\'"
    ]
}
```
```
{
    "message": "Expression `\\'foo\\\\\\'` is missing the following closing delimiter: `'`.\n9:15 title = 'foo\\'",
    "code": "invalid_search_filter",
    "type": "invalid_request",
    "link": "https://docs.meilisearch.com/errors#invalid_search_filter"
}
```

**Expected behavior**
Single `test_1` document in results.

**Meilisearch version:**
Tested with:
- v0.28.1
- v1.0.0
- v1.3.2

**Additional context**
Executing search with filter `foo\\` to get `test_2` document works fine.

POST `/indexes/test/search`
```
{
    "filter": [
        "title = \"foo\\\\\""
    ]
}
``` 
The result returned.
```
{
    "hits": [
        {
            "title": "foo\\\\",
            "id": "test_2"
        }
    ],
    "query": "",
    "processingTimeMs": 0,
    "limit": 20,
    "offset": 0,
    "estimatedTotalHits": 1
}
```

**Comment by Kerollmops** [comment]
Hey @benjamin-cizej 👋

Thank you very much for this excellent report!
We will work on that tomorrow and fix this long-running bug once and for all 🤞

**Comment by benjamin-cizej** [comment]
A bit more info to this - it looks like every odd number of backslashes at the end of the value returns an error.

```
"filter": ["title = 'foo\\'"] // Does not work "foo\"
"filter": ["title = 'foo\\\\'"] // Works "foo\\"
"filter": ["title = 'foo\\\\\\'"] // Does not work "foo\\\"
"filter": ["title = 'foo\\\\\\\\'"] // Works "foo\\\\"
```

**Comment by curquiza** [comment]
Fixed by #4038
Will be released in v1.3.3 this week or next week

**Comment by Kerollmops** [comment]
Hey @benjamin-cizej 👋

The v1.3.3 release is out. You can find it [on the release page](https://github.com/meilisearch/meilisearch/releases/tag/v1.3.3).

Thank you again for the bug report 🧇

**Comment by benjamin-cizej** [comment]
Awesome, I tested and it works perfectly, thanks for a very fast fix!

**Comment by irevoire** [comment]
Hey @benjamin-cizej, at the time, we didn’t realize that this change was a breaking change and thus will have to revert it until the v1.4 next week.
This week, we'll release the v1.3.5 that will revert this bug fix, so my tip would be to stay on the v1.3.4 until the v1.4 comes out 😁 

You can read more about this decision here: https://github.com/meilisearch/meilisearch/issues/4064
Let me know if you need more info!
